relay/nakama/handlers.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	errors2 "errors"
+	"github.com/heroiclabs/nakama-common/runtime"
+	"github.com/rotisserie/eris"
+	"io"
+	"net/http"
+	"pkg.world.dev/world-engine/relay/nakama/allowlist"
+	"pkg.world.dev/world-engine/relay/nakama/errors"
+	"pkg.world.dev/world-engine/relay/nakama/persona"
+	"pkg.world.dev/world-engine/relay/nakama/receipt"
+	"pkg.world.dev/world-engine/relay/nakama/utils"
+	"strings"
+)
+
+// handleClaimPersona handles a request to Nakama to associate the current user with the persona tag in the payload.
+func handleClaimPersona(verifier *persona.Verifier, notifier *receipt.Notifier) nakamaRPCHandler {
+	return func(
+		ctx context.Context,
+		logger runtime.Logger,
+		db *sql.DB,
+		nk runtime.NakamaModule,
+		payload string,
+	) (string, error) {
+		ptr := &persona.StorageObj{}
+		if err := json.Unmarshal([]byte(payload), ptr); err != nil {
+			return utils.LogErrorWithMessageAndCode(
+				logger,
+				err,
+				errors.InvalidArgument,
+				"unable to unmarshal payload: %v",
+				err)
+		}
+
+		result, err := persona.ClaimPersona(
+			ctx,
+			nk,
+			verifier,
+			notifier,
+			ptr,
+			globalCardinalAddress,
+			globalNamespace,
+			&globalPersonaTagAssignment,
+		)
+		if err == nil {
+			return utils.MarshalResult(logger, result)
+		}
+
+		switch {
+		case errors2.Is(eris.Cause(err), persona.ErrPersonaTagStorageObjNotFound):
+			return utils.LogErrorWithMessageAndCode(logger, err, errors.NotFound, "persona tag storage object not found")
+		case errors2.Is(err, persona.ErrPersonaTagEmpty):
+			return utils.LogErrorWithMessageAndCode(
+				logger,
+				err,
+				errors.InvalidArgument,
+				"claim persona tag request must have personaTag field",
+			)
+		}
+		return utils.LogError(logger, err, errors.Internal)
```

Comment:
we actually don't want to use `Internal` as it ends up mapping to `http 500` and crashes webGL clients

Reply:
All right, looks like (4) was right on the ball